src/util/generate_root.c

```diff
@@ -0,0 +1,73 @@
+/******************************************************************************
+ *
+ * Copyright 2018 Xaptum, Inc.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License
+ *
+ *****************************************************************************/
+#include <string.h>
+#include <xtt.h>
+
+int generate_root(struct cli_params* params)
+{
+    const char *privkey_filename = params->privkey;
+    const char *pubkey_filename = params->pubkey;
+    const char *id_filename = params->id;
+    char *root_pub_file = "root_pub.bin";
+    char *root_priv_file = "root_priv.bin";
+    xtt_ecdsap256_pub_key pub;
+    xtt_ecdsap256_priv_key priv;
+    xtt_certificate_root_id root_id;
+
+    unsigned char idpubkey[sizeof(xtt_ecdsap256_pub_key)+sizeof(xtt_certificate_root_id)];
```

Comment:
We should document this format for the root file in the header.

And, maybe let's have two new functions that (de)serialize "root public info" (or call it something else that makes more sense). Something like:
```
typedef struct {unsigned char data[81];} xtt_root_public_info;

/*
 * Serialize the provided root public information into the provided buffer.
 */
void serialize_root_public_info(struct xtt_ecdsap256_pub_key *pub_key, struct xtt_certificate_root_id *id, struct xtt_root_public_info *info_out);
```

At some point (when the `certificates.h` file gets cleaned-up...), some of this may go into the `certificate.h` file, but for now we can keep it in this file (and maybe re-name this file `root.h` or something like that?).